src/components/TodoItem.tsx

```diff
@@ -0,0 +1,115 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  useContext,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodoContext } from './TodoContext';
+
+type Props = { toDo: Todo };
+
+export const TodoItem:React.FC<Props> = ({ toDo }) => {
+  const { updateTodo, deleteTodo } = useContext(TodoContext);
+  const [switchEditTodo, setSwitchEditTodo] = useState(false);
+  const [newTitle, setNewTitle] = useState('' || toDo.title);
```

Comment:
```suggestion
  const [newTitle, setNewTitle] = useState(toDo.title);
```
initial state will always be assigned as todo.title since empty string is false